var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -99,14 +101,28 @@ class Gromacs(CMakePackage, CudaPackage):
         when="@2022: +cuda+mpi",
         description="Enable multi-GPU FFT support with cuFFTMp",
     )
+
     variant(
         "heffte",
         default=False,
         when="@2021: +sycl+mpi",
         description="Enable multi-GPU FFT support with HeFFTe",
     )
+    variant(
+        "rocfft",
+        default=False,
+        when="@2022: +sycl +rocm",
+        description="Enable GPU FFT support with rocFFT",
+    )
+
     variant("opencl", default=False, description="Enable OpenCL support")
     variant("sycl", default=False, when="@2021:", description="Enable SYCL support")
+    variant(
+        "sycl",
+        default=True,
+        when="@2022: +rocm",
+        description="Enable ROCm support when using SYCL",
+    )
```

Comment:
I'm surprised this works when combined with the `variant` declaration just above this - was it intended to keep both? Overall, the constraints as given are not precisely aligned: `@2021+sycl~rocm` has no effect

If you want to force `+sycl` when you have `+rocm`, you can say `conflicts(~sycl, when=+rocm)`